bokeh/plotting/helpers.py

```diff
@@ -200,6 +201,17 @@ def _get_range(range_input):
     raise ValueError("Unrecognized range input: '%s'" % str(range_input))
 
 
+def _get_scale(range_input, axis_type):
+    if isinstance(range_input, (DataRange1d, Range1d)) and axis_type in ["linear", "datetime", "auto"]:
+        return LinearScale()
+    elif isinstance(range_input, (DataRange1d, Range1d)) and axis_type == "log":
+        return LogScale()
+    elif isinstance(range_input, FactorRange):
```

Comment:
The current behavior to allow:

```python
figure(x_range=FactorRange(), x_axis_type="log") #or any axis type, i.e. linear/datetime
```

Though it may barf on the BokehJS side. Should we prevent users from getting into that state?